sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/feedranges/FeedRangeInternal.java

```diff
@@ -19,14 +26,19 @@
 
 import java.io.IOException;
 import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
 import java.util.Base64;
+import java.util.Collections;
 import java.util.List;
 
+import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkArgument;
 import static com.azure.cosmos.implementation.guava25.base.Preconditions.checkNotNull;
 
 @JsonDeserialize(using = FeedRangeInternalDeserializer.class)
 public abstract class FeedRangeInternal extends JsonSerializable implements FeedRange {
     private final static Logger LOGGER = LoggerFactory.getLogger(FeedRangeInternal.class);
+    private final static Long UINT64_TO_DOUBLE_MASK = Long.parseUnsignedLong("9223372036854775808");
+    private final static Long UINT_MAX_VALUE = Long.parseUnsignedLong("4294967295");
```

Comment:
magic numbers, I wonder if there is anyway we can infer the numbers from java constants?

we have the guava primitive types shaded in our code available to be used. I wonder if any of their code can be used for unsigned int 64, etc.

Please see here:
https://github.com/Azure/azure-sdk-for-java/blob/master/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/guava25/primitives/UnsignedLongs.java

the same goes for the rest of this file. I haven't verified if the class from guava has the functionality we need or not. it is worth checking.

Reply:
The magic numbers are special - and cannot be inferred form the MAX_VALUE consts etc. because of the difference in endianness. I could probably write a method to change the endianness and then apply it to the consts - but I don't think it will make the code much easier to read. And however we do it - this will stay code you only touch if you have to anyway :-) I think the test coverage I have in place (checking the calculation of the splits against a .Net baseline is keeping us safe form unintentional regressions?

Reply:
thanks for explanation. LGTM.